src/euphorie/client/browser/training.py

```diff
@@ -12,43 +19,207 @@ class TrainingSlide(BrowserView):
+            if isinstance(existing_measures, dict):
+                existing_measures = [(k, v) for (k, v) in existing_measures.items()]
+            existing_measures = [
+                text.strip() for (text, active) in existing_measures if active
+            ]
+        except:
+            existing_measures = []
+        return existing_measures
+
+    @property
+    @memoize
+    def image(self):
+        if self.is_custom:
+            if getattr(self.context, "image_data", None):
+                _view = self.context.__of__(
+                    self.webhelpers.traversed_session.aq_parent["custom-risks"]
+                ).restrictedTraverse("image-display")
+                _view.field_name = "image_large"
+                return _view.get_or_create_image_scaled()
+            return None
+        image = self.zodb_elem.image and self.zodb_elem.image.data or None
+        if image and self.for_download:
+            try:
+                scales = self.zodb_elem.restrictedTraverse("images", None)
+                if self.item_type == "module":
+                    scale_name = "training_title"
+                else:
+                    scale_name = "training"
+                if scales:
+                    scale = scales.scale(fieldname="image", scale=scale_name)
+                    if scale and scale.data:
+                        image = scale.data.data
+            except:
+                image = None
+        return image
+
+    def slides(self):
+        slides = [{"slide_type": self.item_type, "slide_template": self.slide_template}]
+        if self.existing_measures:
+            slides.append(
+                {
+                    "slide_type": "risk_measures",
+                    "content": self.existing_measures,
+                    "slide_template": "template-measures",
+                }
+            )
+        return slides
+
+
+class TrainingView(BrowserView, survey._StatusHelper):
     """ The view that shows the main-menu Training module
     Currently not active in default Euphorie
     """
+
     variation_class = "variation-risk-assessment"
+    skip_unanswered = False
 
     @property
     @memoize
     def webhelpers(self):
         return self.context.restrictedTraverse("webhelpers")
 
+    @property
+    @memoize
+    def session(self):
+        """ Return the session for this context/request
+        """
+        return self.context.session
+
+    @property
+    @memoize
+    def for_download(self):
+        return "for_download" in self.request
+
+    @property
+    @memoize
+    def title_image(self):
+        survey = self.context.aq_parent
+        return (
+            getattr(survey, "external_site_logo", None)
+            and survey.external_site_logo.data
+            or None
+        )
+
+    @property
+    @memoize
+    def slide_data(self):
+        modules = self.getModulePaths()
+        risks = self.getRisks(modules, skip_unanswered=self.skip_unanswered)
```

Comment:
Looks strange to pass an instance attribute (`self.skip_unanswered`) to an instance method.